### PR TITLE
feat: add history page, search video, filter , video listing on singl…

### PR DESCRIPTION
**Pull request by Bhakti199**
…e video page and modal

- here I added  #11 #14  :
- modal
- search video
- filters video listing on single video page

Screenshot (1365)

### CODE REVIEW

File: src/Components/Modal/Modal.css [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.modal-container {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
```
Comment (tushartiwari7) [inline src/Components/Modal/Modal.css]: There's a shorthand for this, `inset: 0` which will do the same.

File: src/Context/MainContext/SidebarManagementUtils.js [hunk 2]
```diff
@@ -39,3 +39,19 @@ export const removeFromPlayList = (playList, title, videoToRemove) => {
...
+    case "Comedy":
+      return videoList.filter((video) => video.categoryName === "Comedy");
+    case "Action":
+      return videoList.filter((video) => video.categoryName === "Action");
+    case "Horror":
+      return videoList.filter((video) => video.categoryName === "Horror");
+    case "Thriller":
+      return videoList.filter((video) => video.categoryName === "Thriller");
+    case "All":
+      return videoList;
```
Comment (tushartiwari7) [inline src/Context/MainContext/SidebarManagementUtils.js]: Suggestion: you can do this without using switch too in just one line by comparing the `video.categoryName` with `currentCategory`.
```javascript
 return videoList.filter((video) => video.categoryName === currentCategory);
 ```

File: src/Pages/SingleVideoPage/SingeVideoPage.jsx [hunk 3]
```diff
@@ -1,59 +1,120 @@
...
+  const { state, searchInput, dispatch } = useMainContext();
+  let video = state.videoList.find((item) => item._id === videoId.videoId);
+
   return (
     <>
       <div className="single-video-page flex">
         <div className="video-iframe flex">
           <iframe
-            src={`https://www.youtube.com/embed/${videoId.videoId}`}
+            src={`https://www.youtube.com/embed/${videoId.videoId}?autoplay=1`}
```
Comment (tushartiwari7) [inline src/Pages/SingleVideoPage/SingeVideoPage.jsx:26]: This is nice.

File: src/Pages/VideoListingPage/VideoListingPage.css [hunk 4]
```diff
@@ -41,10 +41,14 @@
...
 }
-.category-chips:hover {
+.chip-default {
   background-color: var(--purple-medium);
   color: var(--white-color);
 }
+/* .category-chips:hover {
+  background-color: var(--purple-medium);
+  color: var(--white-color);
+} */
```
Comment (tushartiwari7) [inline src/Pages/VideoListingPage/VideoListingPage.css]: remove unused code.

**Review by tushartiwari7 — APPROVED** [review]
Functionalities are all good.
Just have a look at my suggestions and you are good to merge.

File: src/Components/Modal/Modal.jsx [hunk 5]
```diff
@@ -1,9 +1,85 @@
...
+                },
+              });
+              setOpenModal(false);
+            }}
+          >
+            <MdAddCircle size="25" />
+          </span>
+        </div>
+        <div className="playlist-container">
+          {state.playList.map((element, index) =>
```
Comment (JantuDeb) [inline src/Components/Modal/Modal.jsx]: Use proper name for variable

File: src/Pages/SingleVideoPage/SingeVideoPage.jsx [hunk 6]
```diff
@@ -1,59 +1,120 @@
...
-import { Sidebar } from "../../Components/Index";
+import { Sidebar, VideoCard, Modal } from "../../Components/Index";
 export const SingeVideoPage = () => {
+  const [singlePageModal, setSinglePageModal] = useState(false);
   const videoId = useParams();
-  const { videos, state } = useMainContext();
-  let video = videos.find((item) => item._id === videoId.videoId);
- 
+  const { state, searchInput, dispatch } = useMainContext();
+  let video = state.videoList.find((item) => item._id === videoId.videoId);
```
Comment (JantuDeb) [inline src/Pages/SingleVideoPage/SingeVideoPage.jsx:19]: You can store the video I'd in video 
video{
videoId:""

}

**Review by JantuDeb — APPROVED** [review]
Overall code looks structured. I didn't find any major issues with the code.

**Pull request merged**